tests/30rooms/30history-visibility.pl

```diff
@@ -533,6 +534,43 @@ sub test_history_visibility
       });
    };
 
+test "Backfill works correctly with history visilibty set to joined",
+   requires => [ magic_local_user_and_room_fixtures( with_alias => 1), local_user_fixture(), remote_user_fixture() ],
+
+   do => sub {
+      my ( $user, $room_id, $room_alias, $another_user, $remote_user, $room_alias_name ) = @_;
+
+      matrix_set_room_history_visibility( $user, $room_id, "joined" )
+      ->then( sub {
+         repeat( sub {
+            my $msgnum = $_[0];
+
+            matrix_send_room_text_message( $user, $room_id, body => "Message $msgnum" );
+         }, foreach => [ 1 .. 100 ]);
+      })->then( sub {
+         matrix_join_room( $another_user, $room_alias );
```

Comment:
what's the point of this?

Reply:
I've added a comment, but the bug only manifests if there is a state event in the chunk. (Initially I thought it also had to appear in the auth chain, hence using a join, but I don't think that's true)